plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/ui/internal/view/CDKActionProvider.java

```diff
@@ -89,8 +84,11 @@ protected Object adaptToViewItem(IServer server) {
 			if( beh != null ) {
 				adb = (ServiceManagerEnvironment)beh.getSharedData(ServiceManagerEnvironment.SHARED_INFO_KEY);
 			}
-			if( adb != null )
-				return new CDKOpenshiftUtility().findExistingOpenshiftConnection(server, adb);
+			
+			if( adb != null ) {
+				OpenshiftUIModel model = new OpenshiftUIModel();
+				return model.getConnectionWrapperForConnection(new CDKOpenshiftUtility().findExistingOpenshiftConnection(server, adb));
```

Comment:
yes I see the problem that you're pointing at and went unnoticed so far :(
Having singletons would be most likely the more robust option than the manual disposal I think  even though singletons have their own problems